data-prepper-plugins/grok-prepper/README.md

```diff
@@ -16,11 +16,34 @@ prepper:
 
 ## Configuration
 
+TODO: provide examples for using each configuration
+
 * `match` (Optional): A `Map<String, List<String>>` that specifies which fields of a Record to match which patterns against. Default value is `{}`
 
-* `keep_empty_captures` (Optional): A `boolean` that specifies whether `null` captures should be kept. Default value is `false`
 
+* `keep_empty_captures` (Optional): A `boolean` that specifies whether `null` captures should be kept. Default value is `false`
+  
+  
 * `named_captures_only` (Optional): A `boolean` that specifies whether to only keep named captures. Default value is `true`
+  
+  
+* `break_on_match` (Optional): A `boolean` that specifies whether to match all patterns from the `match` field against a Record, 
+  or to stop once the first successful pattern match is found. Default value is `true`
+  
+
+* `overwrite` (Optional): A `List<String>` that specifies which existing fields of a Record to overwrite if there is a capture with the same field value. Default value is `[]`
```

Comment:
I wonder if we should take a different approach for this configuration. 

At the very least, we should change the name.

But, even more so, it seems odd because for this to have any meaning, you need to have the same field in both `match` and `overwrite`.

Reply:
Also this field is most commonly used to overwrite the original `message` to save space. However, the implementation does not actually require all overwrite keys to be in `match`. If a key is in the overwrite list, then a capture with the same key value as a previous capture will be overwritten.